### PR TITLE
UI, DataTable: #38429, typecheck for Icons in Columns

### PR DESCRIPTION
**Pull request by nhaagen**
https://mantis.ilias.de/view.php?id=38429

### CODE REVIEW

**Comment by tfamula** [comment]
Hi @nhaagen!

Thanks for the PR!
Do I see it correctly that in future only the `Icon` Component is allowed for the `StatusIcon` Column?
Please note, that this Column type already has some usages since ILIAS 9 (e.g. Skill Management, but also other components), which have to be adapted.
Moreover, I would be happy about unit tests which test the new `Boolean::format()` and `StatusIcon::format()` methods.

**Comment by nhaagen** [comment]
Hi @tfamula ,
it's just been very recent that Colum::format may return Components. 
I really hope to not cause too much adaption trouble in enforcing the icons, since I think it's a valuable improvement over pre-rendering.
I added some tests, too.

**Review by klees — CHANGES_REQUESTED** [review]
Hi @nhaagen,

LGTM, mostly. At first I wondered if we would really want to have flexible icons for true/false, but then I realized that a boolean could express various states, e.g. "happy", where we could use smiley faces or something. Then this occured to me:

* [x] Glyph: add `Glyph` as another option for the `Boolean` column.

That's it.

Thanks!

**Comment by nhaagen** [comment]
Screenshot from 2024-01-11 09-23-03

**Comment by klees** [comment]
@Amstutz, @thibsy Would you be so kind and move this on? Thanks!

**Comment by oliversamoila** [comment]
Hello @klees , 
can you briefly explain to me why we need a glyph column as a representation of booleans?
The boolean indicates a value - like in the example yes/no , 0/1, cross, tick. So far so good. I also understand that there are symbols that represent this well.
But the glyph would serve to trigger an underlying action. With a value pair in the sense of a boolean, I would then expect the function of a toggle. 
Is that the case? If so, shouldn't it rather be a toggle labelled with two glyphs? And if not, what do we do with the Boolean glyph?

Thank you very much in advance.

**Comment by klees** [comment]
Hi @oliversamoila,

no, we do not intent to introduce a toggle in disguise here.

The glyphs are currently designed to include an action, as you mention. We (i.e. the UI coordinators) found this to problematic on various occasions: Bulky Buttons with Glyphs could have had two actions. We had to jump extra hoops for a11y, although we already solved problems for buttons. Glyphs behave strangely e.g. in the Pagination. We are currently looking to remove the action from the Glyphs and shift it to the Buttons instead. #6922 shows the current code. Should come out a lot cleaner once we have done the move.

So basically, we want to the symbolism from the glyphs (booleans could stand for e.g. sad/happy) but not the actions.

@nhaagen: Could you clarify that in the description and then check programmatically if the provided actions do indeed carry no action?

Hope this makes sense...

Kind regards!

**Review by klees — APPROVED** [review]

**Comment by tfamula** [comment]
Fix usages in Skill Management: https://github.com/ILIAS-eLearning/ILIAS/commit/38f1f568a6fd53a3b5d3d9c791719bb6a370c8da
